### ISSUE TITLE
LZMAError('Compressed data ended before the end-of-stream marker was reached')

### ISSUE THREAD

**Issue by Moult** [mgorny/eclean-kernel#29]
Thought I'd try this out and I came across:

```
$ eclean-kernel -l
eclean-kernel has met the following issue:

  LZMAError('Compressed data ended before the end-of-stream marker was reached')

If you believe that the mentioned issue is a bug, please report it
to https://github.com/mgorny/eclean-kernel/issues. If possible,
please attach the output of 'eclean-kernel --list-kernels' and your
regular eclean-kernel call with additional '--debug' argument.
```

```
➜  ~ eclean-kernel -lD
DEBUG:root:Sorter: <ecleankernel.sort.VersionSort object at 0x7fbda67a0f10>
DEBUG:root:Layout failed: <class 'ecleankernel.layout.blspec.BlSpecLayout'>; exception: /boot/[EFI/]18b34b2f29fbc54b6dd357085a1b8972 not found
DEBUG:root:Layout: <ecleankernel.layout.std.StdLayout object at 0x7fbda67cd130>
DEBUG:root:Bootloader failed: <class 'ecleankernel.bootloader.lilo.LILO'>
DEBUG:root:/boot/grub/grub.cfg found
DEBUG:root:Bootloader: <ecleankernel.bootloader.grub2.GRUB2 object at 0x7fbda67cd160>
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.9/eclean-kernel", line 33, in <module>
    sys.exit(load_entry_point('eclean-kernel==2.99.3', 'console_scripts', 'eclean-kernel')())
  File "/usr/lib/python3.9/site-packages/ecleankernel/__main__.py", line 382, in setuptools_main
    sys.exit(main(sys.argv[1:]))
  File "/usr/lib/python3.9/site-packages/ecleankernel/__main__.py", line 227, in main
    kernels = layout.find_kernels(exclusions=exclusions)
  File "/usr/lib/python3.9/site-packages/ecleankernel/layout/std.py", line 94, in find_kernels
    kobj = KernelImage(path)
  File "/usr/lib/python3.9/site-packages/ecleankernel/file.py", line 97, in __init__
    self.internal_version = self.read_internal_version()
  File "/usr/lib/python3.9/site-packages/ecleankernel/file.py", line 160, in read_internal_version
    b = self.decompress_raw()
  File "/usr/lib/python3.9/site-packages/ecleankernel/file.py", line 135, in decompress_raw
    return getattr(mod, 'decompress')(f.read())
  File "/usr/lib/python3.9/lzma.py", line 350, in decompress
    raise LZMAError("Compressed data ended before the "
_lzma.LZMAError: Compressed data ended before the end-of-stream marker was reached
```

**Comment by natanhp** [comment]
Hi, maybe you can check out my fork on https://github.com/natanhp/eclean-kernel/tree/fix-lzma-decompress for a temporary solution.

**Comment by xgqt** [comment]
> Hi, maybe you can check out my fork on https://github.com/natanhp/eclean-kernel/tree/fix-lzma-decompress for a temporary solution.

this looks ok, care to maeke a PR?

**Comment by natanhp** [comment]
Yeah, sure

**Comment by mgorny** [comment]
Fixed via #30.

**Comment by AstrocalcVB** [comment]
How long time before we will see a release with this fix? As it stands now 2.99.3 appears broken to me.

**Comment by mgorny** [comment]
I'm sorry about the delay. I'm working on adding EFIstub support today, and I will make a release once I'm done.
